frontend/src/components/Form.tsx

```diff
@@ -186,6 +187,37 @@ export const Form: React.FunctionComponent<FormProps> = ({
         setDisableDownloadButton(false);
         setProgress(100);
       }
+      if (endpoint === "download-txt") {
+        setProgress(30);
+        setLoading(false);
+        const rawResponse = await axios({
+          url: module.apiEndpoint + `download-txt`,
+          method: "POST",
+          responseType: "blob",
+          data: data,
+          headers: {
+            "Content-Type": "application/json",
+          },
+        });
+        const url = window.URL.createObjectURL(new Blob([rawResponse.data]));
+        const link = document.createElement("a");
+        link.href = url;
+        link.setAttribute(
+          "download",
+          `${data.database}_${molecules}_${data.min_wavenumber_range}_${data.max_wavenumber_range}cm-1_${data.tgas}K_${data.pressure}atm.csv`
+        );
+        document.body.appendChild(link);
+        link.click();
+        setDisableDownloadButton(false);
+        const response = await rawResponse.data;
+        if (response.error) {
+          handleBadResponse(response.error);
+        } else {
+          setDisableDownloadButton(false);
+        }
+        setDisableDownloadButton(false);
+        setProgress(100);
```

Comment:
There's too much duplication between this and download-spectrum. Could we find a way to refactor it to have more code re-use?